Reading of Field_data from VTK not working

When I load the following file with meshio 2.3.6:

    # vtk DataFile Version 3.0
    Unstructured Grid
    ASCII
    DATASET UNSTRUCTURED_GRID

    FIELD TimesAndCycles 2
    TIME 1 1 double
    1.728000000000e+05
    CYLCE 1 1 long
    2

    POINTS 4 double
    -2.000000000000e+01 0.000000000000e+00 5.200000000000e+01
    -2.000000000000e+01 0.000000000000e+00 5.205000000000e+01
    -1.975000000000e+01 0.000000000000e+00 5.205000000000e+01
    -1.975000000000e+01 0.000000000000e+00 5.200000000000e+01

    CELLS 1 5
    4 0 1 2 3

    CELL_TYPES 1
    9

    POINT_DATA 4
    SCALARS HEAD double 1
    LOOKUP_TABLE default
    6.600000000000e-01
    6.600000000000e-01
    6.592496815932e-01
    6.592496815932e-01
    SCALARS CONCENTRATION1 double 1
    LOOKUP_TABLE default
    0.000000000000e+00
    0.000000000000e+00
    3.426485920483e-105
    1.145624884345e-105

    CELL_DATA 1
    SCALARS VELOCITY1_X double 1
    LOOKUP_TABLE default
    3.001273627178e-07
    SCALARS VELOCITY1_Y double 1
    LOOKUP_TABLE default
    -2.287059430728e-18
    SCALARS VELOCITY1_Z double 1
    LOOKUP_TABLE default
    0.000000000000e+00

I get the following error:

    /usr/local/lib/python3.6/dist-packages/meshio/vtk_io.py in read_buffer(f)
        230             else:
        231                 assert active == "CELL_DATA", "Illegal FIELD in section '{}'.".format(
    --> 232                     active
        233                 )
        234                 d = cell_data_raw

    AssertionError: Illegal FIELD in section 'None'.

But the vtk python package as well as Paraview are able to read the Field_data correctly. Is there some way, that meshio can handle it?

--- a/meshio/vtk_io.py
+++ b/meshio/vtk_io.py
@@ -134,11 +134,11 @@ def read_buffer(f):
     c = None
     ct = None
 
-    # One of the problem in reading VTK files are POINT_DATA and CELL_DATA
-    # fields. They can contain a number of SCALARS+LOOKUP_TABLE tables, without
-    # giving and indication of how many there are. Hence, SCALARS must be
-    # treated like a first-class section. To associate it with POINT/CELL_DATA,
-    # we store the `active` section in this variable.
+    # One of the problem in reading VTK files are POINT_DATA and CELL_DATA fields. They
+    # can contain a number of SCALARS+LOOKUP_TABLE tables, without giving and indication
+    # of how many there are. Hence, SCALARS must be treated like a first-class section.
+    # To associate it with POINT/CELL_DATA, we store the `active` section in this
+    # variable.
     active = None
 
     while True:
@@ -189,41 +189,7 @@ def read_buffer(f):
             data = numpy.fromfile(f, count=num_items * 4, sep=" ", dtype=float)
             rgba = data.reshape((num_items, 4))  # noqa F841
 
-        elif section == "SCALARS":
-            if active == "POINT_DATA":
-                d = point_data
-            else:
-                assert active == "CELL_DATA", "Illegal SCALARS in section '{}'.".format(
-                    active
-                )
-                d = cell_data_raw
-
-            d.update(_read_scalar_field(f, num_items, split))
-
-        elif section == "VECTORS":
-            if active == "POINT_DATA":
-                d = point_data
-            else:
-                assert active == "CELL_DATA", "Illegal SCALARS in section '{}'.".format(
-                    active
-                )
-                d = cell_data_raw
-
-            d.update(_read_vector_field(f, num_items, split))
-
-        elif section == "TENSORS":
-            if active == "POINT_DATA":
-                d = point_data
-            elif active == "CELL_DATA":
-                d = cell_data_raw
-            else:
-                d = field_data
-
-            d.update(_read_tensor_field(f, num_items, split))
-
         else:
-            assert section == "FIELD", "Unknown section '{}'.".format(section)
-
             if active == "POINT_DATA":
                 d = point_data
             elif active == "CELL_DATA":
@@ -231,7 +197,15 @@ def read_buffer(f):
             else:
                 d = field_data
 
-            d.update(_read_fields(f, int(split[2]), is_ascii))
+            if section == "SCALARS":
+                d.update(_read_scalar_field(f, num_items, split))
+            elif section == "VECTORS":
+                d.update(_read_vector_field(f, num_items, split))
+            elif section == "TENSORS":
+                d.update(_read_tensor_field(f, num_items, split))
+            else:
+                assert section == "FIELD", "Unknown section '{}'.".format(section)
+                d.update(_read_fields(f, int(split[2]), is_ascii))
 
     assert c is not None, "Required section CELLS not found."
     assert ct is not None, "Required section CELL_TYPES not found."
@@ -303,7 +277,7 @@ def _read_scalar_field(f, num_data, split):
 
 def _read_vector_field(f, num_data, split):
     data_name = split[1]
-    data_type = split[2]
+    data_type = split[2].lower()
 
     dtype = numpy.dtype(vtk_to_numpy_dtype_name[data_type])
     data = numpy.fromfile(f, count=3 * num_data, sep=" ", dtype=dtype).reshape(-1, 3)

--- a/meshio/vtk_io.py
+++ b/meshio/vtk_io.py
@@ -214,11 +214,10 @@ def read_buffer(f):
         elif section == "TENSORS":
             if active == "POINT_DATA":
                 d = point_data
-            else:
-                assert active == "CELL_DATA", "Illegal SCALARS in section '{}'.".format(
-                    active
-                )
+            elif active == "CELL_DATA":
                 d = cell_data_raw
+            else:
+                d = field_data
 
             d.update(_read_tensor_field(f, num_items, split))
 
@@ -227,11 +226,10 @@ def read_buffer(f):
 
             if active == "POINT_DATA":
                 d = point_data
-            else:
-                assert active == "CELL_DATA", "Illegal FIELD in section '{}'.".format(
-                    active
-                )
+            elif active == "CELL_DATA":
                 d = cell_data_raw
+            else:
+                d = field_data
 
             d.update(_read_fields(f, int(split[2]), is_ascii))
 

